Packs/RedCanary/Integrations/RedCanary/RedCanary.py

```diff
@@ -539,24 +539,34 @@ def execute_playbook(playbook_id, detection_id):
 
 def fetch_incidents():
     last_run = demisto.getLastRun()
-    if last_run and 'time' in last_run:
+    last_incidents_ids = []
+
+    if last_run:
         last_fetch = last_run.get('time')
         last_fetch = datetime.strptime(last_fetch, TIME_FORMAT)
+        last_incidents_ids = last_run.get('last_event_ids')
     else:
+        # first time fetching
         last_fetch = parse_date_range(demisto.params().get('fetch_time', '3 days'), TIME_FORMAT)[0]
 
     LOG('iterating on detections, looking for more recent than {}'.format(last_fetch))
     incidents = []
+    new_incidents_ids = []
     for raw_detection in get_unacknowledged_detections(last_fetch, per_page=2):
         LOG('found detection #{}'.format(raw_detection['id']))
```

Comment:
maybe move this line after the `if incident_id not in last_incidents_ids:`
and:
```suggestion
        LOG('found a new detection in RedCanary #{}'.format(raw_detection['id']))
```

Reply:
I'm using it later in the outer scop